pkg/syncer/fullsync.go

```diff
@@ -467,15 +497,54 @@ func volumeInfoCRFullSync(ctx context.Context, metadataSyncer *metadataSyncInfor
 					// Add volume to deletion map.
 					volumeInfoCrDeletionMap[vc][cnsvolumeinfo.Spec.VolumeID] = true
 				}
+			} else {
```

Comment:
I don't think we should mix the conditionals here. For us we are concerned if there is a mismatch between cns and cnsvolumeinfo aggregated snapshot size, we don't really care if the PV exists or not. We simply want to reflect the right quota.